pkg/controllers/cassandra/service/control.go

```diff
@@ -34,9 +37,17 @@ func NewControl(
 
 func (e *defaultCassandraClusterServiceControl) Sync(cluster *v1alpha1.CassandraCluster) error {
 	svc := ServiceForCluster(cluster)
-	_, err := e.kubeClient.CoreV1().Services(svc.Namespace).Update(svc)
+	client := e.kubeClient.CoreV1().Services(svc.Namespace)
+	existingSvc, err := e.serviceLister.Services(cluster.Namespace).Get(svc.Name)
 	if k8sErrors.IsNotFound(err) {
-		_, err = e.kubeClient.CoreV1().Services(svc.Namespace).Create(svc)
+		_, err := client.Create(svc)
+		return err
+	} else if err != nil {
```

Comment:
Can get rid of the `else` part here - if the error isn't not found, the next block will then check if it's not nil, and return err if it isn't nil 😄

Reply:
done.